test/test.transporters.js

```diff
@@ -124,4 +124,22 @@ it('should return errors within response body as instances of Error', function(d
       done();
     });
   });
+
+  it('should handle 5xx responses that include errors', function(done) {
+    var google = require('../lib/googleapis');
+    var scope = nock('https://www.googleapis.com')
+      .post('/urlshortener/v1/url')
+      .reply(500, { error: {message: 'There was an error!'}});
```

Comment:
The JSON I pasted (which you have above) is the complete JSON representation of the response object. So only the 'body' part is the JSON returned in the response body.